docs/deployment.md

```diff
@@ -0,0 +1,16 @@
+# Deployment From PR to Production
+
+1. Create a PR. Get it reviewed.
+2. **Make sure master is merged into the PR branch** and make sure all CI tests pass.
+3. Deploy to your QA slack and app environment first. (Something like `git push qa my_pr_branch:master`)
+4. Watch the logs and manually tests your changes in QA.
+5. Once you're satisfied that it's safe deploy the PR branch to PROD. (Something like `git push production my_pr_branch:master`)
+6. Watch the logs and manually tests your changes in PROD.
+7. If all goes well, then merge your branch into master and you're done!
+
+## Rollback
+If something horrible happens the `git push -f production master`
```

Comment:
This seems sloppy and I also worry about migrations issues.

Reply:
Not with migrations that I've ever seen... So let's give it a go. You could maybe do some tricky scripting, something like: `python manage.py migrate my_app $(ls -t my_app/migrations/ | head -1 | cut -d . -f 1)` This should target the most recent migration, not sure if this would work in your deployment, does GIT copy file modification dates on clone? You could always just sort by filename instead. But beware then you won't pull in migrations from other apps, like if you add a package that has an app it may need migrations. Also, if your migrations are non-elidable you're just screwed of course.

Reply:
For now I added docs about how you could by hand. To automate this, I think I would do some sort of filename-only version of `git diff master` and look for all new migration files.

That said... it might just be better to always do this by hand. Migrations always feel special case enough that automation might just hurt more.